bluepyopt/deapext/tools/eps.c

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright (c) 2016, EPFL/Blue Brain Project
+
+ This file is part of BluePyOpt <https://github.com/BlueBrain/BluePyOpt>
+
+ This library is free software; you can redistribute it and/or modify it under
+ the terms of the GNU Lesser General Public License version 3.0 as published
+ by the Free Software Foundation.
+
+ This library is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with this library; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ The code in this file was original written in 2015 at the                      
+ BlueBrain Project, EPFL, Lausanne                                              
+ The authors were Werner Van Geit, Michael Gevaert and Jean-Denis Courcol       
+ It is based on a C implementation of the IBEA algorithm in the PISA            
+ optimization framework developed at the ETH, Zurich                            
+ http://www.tik.ee.ethz.ch/pisa/selectors/ibea/?page=ibea.php                   
+ 
+*/
+
+#include <Python.h>
+
+static PyObject * eps_indicator(PyObject *self, PyObject *args)
```

Comment:
Would have to look it up, was something I did a year ago.
As far as I know I sped up Giuseppe's code in 3 stages, for in total a speedup of factor 800 or so.
10x was getting rid of numpy, 10x was preventing some copies in memory and the last part was this C code. 
We should anyway add some tests to this code, and we might want to compare to a python implementation.
We could also run Giuseppe synaptic plasticity code with/without it.